diff --git a/adapter/outboundgroup/loadbalance.go b/adapter/outboundgroup/loadbalance.go
index c3222b3..2acaea4 100644
--- a/adapter/outboundgroup/loadbalance.go
+++ b/adapter/outboundgroup/loadbalance.go
@@ -246,6 +246,8 @@ func NewLoadBalance(option *GroupCommonOption, providers []provider.ProxyProvide
 	switch strategy {
 	case "consistent-hashing":
 		strategyFn = strategyConsistentHashing(option.URL)
+	case "round-robin0":
+		strategyFn = strategyRoundRobin0(option.URL)
 	case "round-robin":
 		strategyFn = strategyRoundRobin(option.URL)
 	case "sticky-sessions":
@@ -276,3 +278,15 @@ func NewLoadBalance(option *GroupCommonOption, providers []provider.ProxyProvide
 		Icon:           option.Icon,
 	}, nil
 }
+
+func strategyRoundRobin0(url string) strategyFn {
+	idx := 0
+	idxMutex := sync.Mutex{}
+	return func(proxies []C.Proxy, metadata *C.Metadata, touch bool) (proxy C.Proxy) {
+		idxMutex.Lock()
+		idx = (idx + 1) % len(proxies)
+		proxy = proxies[idx]
+		idxMutex.Unlock()
+		return proxy
+	}
+}
