diff --git a/adapter/outboundgroup/groupbase.go b/adapter/outboundgroup/groupbase.go
index 9e705677..53b92242 100644
--- a/adapter/outboundgroup/groupbase.go
+++ b/adapter/outboundgroup/groupbase.go
@@ -99,7 +99,49 @@ func (gb *GroupBase) Touch() {
 	}
 }
 
-func (gb *GroupBase) GetProxies(touch bool) []C.Proxy {
+
+func (gb *GroupBase) GetProxies(touch bool) (proxies []C.Proxy) {
+	proxies = gb.providerProxies
+	if proxies != nil {
+		return
+	}
+	gb.failedTestMux.Lock()
+	defer gb.failedTestMux.Unlock()
+	proxies = gb.providerProxies
+	if proxies != nil {
+		return
+	}
+	chDirtyCache := make(chan struct{}, 1)
+	chDone := tunnel.World
+	tf := false
+	type follow interface {
+		AddFollower(chan<- struct{})
+	}
+	for _, pd := range gb.providers {
+		if pf, ok := pd.(follow); ok {
+			pf.AddFollower(chDirtyCache)
+			tf = true
+		}
+	}
+	proxies = gb._GetProxies(false)
+	gb.providerProxies = proxies
+	updater := func() {
+		for {
+			select {
+			case <-chDirtyCache:
+				gb.providerProxies = gb._GetProxies(false)
+			case <-chDone:
+				return
+			}
+		}
+	}
+	if tf {
+	
+go updater()
+	}
+	return
+}
+func (gb *GroupBase) _GetProxies(touch bool) []C.Proxy {
 	providerVersions := make([]uint32, len(gb.providers))
 	for i, pd := range gb.providers {
 		if touch { // touch first
diff --git a/adapter/outboundgroup/loadbalance.go b/adapter/outboundgroup/loadbalance.go
index dff9b5ed..d5d59adc 100644
--- a/adapter/outboundgroup/loadbalance.go
+++ b/adapter/outboundgroup/loadbalance.go
@@ -8,6 +8,7 @@ import (
 	"net"
 	"sync"
 	"time"
+	"sync/atomic"
 
 	"github.com/metacubex/mihomo/common/callback"
 	"github.com/metacubex/mihomo/common/lru"
@@ -160,6 +161,14 @@ func strategyRoundRobin(url string) strategyFn {
 	}
 }
 
+
+func strategyRoundRobin0(url string) strategyFn {
+	idx := &atomic.Uint32{}
+	return func(proxies []C.Proxy, metadata *C.Metadata, touch bool) (proxy C.Proxy) {
+		proxy = proxies[idx.Add(1) % (uint32)(len(proxies))]
+		return proxy
+	}
+}
 func strategyConsistentHashing(url string) strategyFn {
 	maxRetry := 5
 	return func(proxies []C.Proxy, metadata *C.Metadata, touch bool) C.Proxy {
@@ -242,6 +251,9 @@ func (lb *LoadBalance) MarshalJSON() ([]byte, error) {
 func NewLoadBalance(option *GroupCommonOption, providers []P.ProxyProvider, strategy string) (lb *LoadBalance, err error) {
 	var strategyFn strategyFn
 	switch strategy {
+
+case "round-robin0":
+	strategyFn = strategyRoundRobin0(option.URL)
 	case "consistent-hashing":
 		strategyFn = strategyConsistentHashing(option.URL)
 	case "round-robin":
diff --git a/adapter/provider/provider.go b/adapter/provider/provider.go
index e5d3b8ac..dfe1206d 100644
--- a/adapter/provider/provider.go
+++ b/adapter/provider/provider.go
@@ -9,6 +9,7 @@ import (
 	"runtime"
 	"strings"
 	"time"
+	"sync"
 
 	"github.com/metacubex/mihomo/adapter"
 	"github.com/metacubex/mihomo/common/convert"
@@ -112,11 +113,38 @@ type ProxySetProvider struct {
 }
 
 type proxySetProvider struct {
+	mutex    sync.Mutex
+	follower map[chan <- struct{}]struct{}
 	baseProvider
 	*resource.Fetcher[[]C.Proxy]
 	subscriptionInfo *SubscriptionInfo
 }
 
+
+func (pp *proxySetProvider) AddFollower(c chan<- struct{})  {
+	pp.mutex.Lock()
+	defer pp.mutex.Unlock()
+	if len(pp.follower) == 0 {
+		pp.follower = make(map[chan <- struct{}]struct{})
+	}
+	pp.follower[c] = struct{}{}
+}
+func (pp *proxySetProvider) UpdateFollower()  {
+	pp.mutex.Lock()
+	defer pp.mutex.Unlock()
+	for c := range pp.follower {
+		select {
+			case c <- struct{}{}:
+				// pass
+			default:
+				// pass
+			}
+	}
+}
+func (pp *proxySetProvider) setProxies(proxies []C.Proxy) {
+	pp.baseProvider.setProxies(proxies)
+	pp.UpdateFollower()
+}
 func (pp *proxySetProvider) MarshalJSON() ([]byte, error) {
 	return json.Marshal(providerForApi{
 		Name:             pp.Name(),
diff --git a/component/dialer/dialer.go b/component/dialer/dialer.go
index d490dca3..79752250 100644
--- a/component/dialer/dialer.go
+++ b/component/dialer/dialer.go
@@ -11,6 +11,8 @@ import (
 	"sync"
 	"syscall"
 	"time"
+	"math/rand"
+
 
 	"github.com/metacubex/mihomo/component/keepalive"
 	"github.com/metacubex/mihomo/component/resolver"
@@ -394,6 +396,10 @@ func parseAddr(ctx context.Context, network, address string, preferResolver reso
 			ips[i] = ip.Unmap()
 		}
 	}
+	if l:= len(ips);l > 1{
+	j := rand.Intn(l)
+	ips[0], ips[j] = ips[j], ips[0]  }
+
 	return ips, port, nil
 }
 
diff --git a/tunnel/tunnel.go b/tunnel/tunnel.go
index fc901eb3..3ab1d8df 100644
--- a/tunnel/tunnel.go
+++ b/tunnel/tunnel.go
@@ -124,7 +124,10 @@ func (t tunnel) RuleUpdateCallback() *utils.Callback[P.RuleProvider] {
 	return ruleUpdateCallback
 }
 
+var World = make(chan struct{})
 func OnSuspend() {
+	close(World)
+	World = make(chan struct{})
 	status.Store(Suspend)
 }
 
